File: onto_generator/src/main/java/com/example/onto_generator/model/ObjectPropertyAxiomsMetrics.java
package com.example.onto_generator.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.RequiredArgsConstructor;

@Builder
@AllArgsConstructor
@RequiredArgsConstructor
@Data
public class ObjectPropertyAxiomsMetrics {
    private int subObjectPropertyAxioms;
    private int equivalentObjectPropertyAxioms;
    private int inverseObjectPropertyAxiomsMetric;
    private int disjointObjectPropertyAxiomsMetric;
    private int functionalObjectPropertyAxiomsMetric;
    private int inverseFunctionalObjectPropertiesAxiomsMetric;
    private int transitiveObjectPropertyAxiomsMetric;
    private int symmetricObjectPropertyAxiomsMetric;
    private int asymmetricObjectPropertyAxiomsMetric;
    private int reflexiveObjectPropertyAxiomsMetric;
    private int irreflexiveObjectPropertyAxiomsMetric;
    private int objectPropertyDomainAxiomsMetric;
    private int objectPropertyRangeAxiomsMetric;
    private int subPropertyChainOfAxiomsMetric;
}

File: frontend/src/app/model/ObjectPropertyAxiomsMetrics.ts
export class ObjectPropertyAxiomsMetrics {
  subObjectPropertyAxioms: number| undefined;
  equivalentObjectPropertyAxioms: number| undefined;
  inverseObjectPropertyAxiomsMetric: number| undefined;
  disjointObjectPropertyAxiomsMetric: number| undefined;
  functionalObjectPropertyAxiomsMetric: number| undefined;
  inverseFunctionalObjectPropertiesAxiomsMetric: number| undefined;
  transitiveObjectPropertyAxiomsMetric: number| undefined;
  symmetricObjectPropertyAxiomsMetric: string| undefined;
  asymmetricObjectPropertyAxiomsMetric: Set<string>| undefined;
  reflexiveObjectPropertyAxiomsMetric: Set<string>| undefined;
  irreflexiveObjectPropertyAxiomsMetric: Set<string>| undefined;
  objectPropertyDomainAxiomsMetric: number| undefined;
  objectPropertyRangeAxiomsMetric: number| undefined;
  subPropertyChainOfAxiomsMetric: number| undefined;
}
